src/phases.ts

```diff
@@ -2670,8 +2670,8 @@ export class MovePhase extends BattlePhase {
         // account for metal burst and comeuppance hitting remaining targets in double battles
         // counterattack will redirect to remaining ally if original attacker faints
         if (this.scene.currentBattle.double && this.move.getMove().hasFlag(MoveFlags.REDIRECT_COUNTER)) {
-          if (!this.scene.getEnemyField()[this.targets[0]]) {
-            this.targets[0] = this.scene.getEnemyField().find(p => p.isActive(true)).getBattlerIndex();
+          if (this.scene.getField()[this.targets[0]].hp === 0) {
+            this.targets[0] = this.scene.getEnemyField().find(p => p.hp > 0)?.getBattlerIndex();
```

Comment:
```suggestion
            const opposingField = pokemon.isPlayer() ? this.scene.getEnemyField() : this.scene.getPlayerField()
            this.targets[0] = opposingField.find(p => p.hp > 0)?.getBattlerIndex();
```
If an enemy used Metal Burst, and the last attacker fainted, the enemy's Metal Burst would be redirected to their ally without this change

Reply:
Metal Burst is supposed to [redirect to the ally if the last attacker faints](https://bulbapedia.bulbagarden.net/wiki/Metal_Burst_(move)#:~:text=However%2C%20if%20the%20last%20Pok%C3%A9mon%20to%20hit%20the%20user%20faints%20before%20the%20user%20moves%2C%20the%20remaining%20enemy%20Pok%C3%A9mon%20will%20be%20hit%20instead%2C%20like%20with%20a%20regular%20single%2Dtarget%20damaging%20move%20(using%20the%20damage%20taken%20from%20the%20move%20of%20the%20now%2Dfainted%20Pok%C3%A9mon%20that%20had%20hit%20the%20user).)